### PR TITLE
Editorconfig and codebase format

### PR DESCRIPTION
**Pull request by aronwk-aaron**
Add editorconfig
Format codebase
Add blame ignore and update contributor info

Tested that the server still compiles and runs

### CODE REVIEW

**Comment by aronwk-aaron** [comment]
Do no Squash Merge

**Comment by EmosewaMC** [comment]
Is this finished? or still waiting?

**Comment by aronwk-aaron** [comment]
Finished and ready for review

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+indent_style=tab
+tab_width=4
+charset=utf-8
+trim_trailing_whitespace=true
+end_of_line=lf
+insert_final_newline=true
+
+[*.{c++,cc,cpp,cxx,h,h++,hh,hpp,hxx,inl,ipp,tlh,tli}]
+
+vc_generate_documentation_comments=doxygen_slash_star
```
Comment (EmosewaMC) [inline .editorconfig:15]: Is this supposed to be slash_star or slash_star_star?  the wording of this setting implies to me it should be slash_star_star but I could be wrong here.

**Review by EmosewaMC — APPROVED** [review]
One minor comment

File: .editorconfig (hunk 1 above)
Comment (aronwk-aaron) [reply .editorconfig:15]: Slash_star
Other options are triple_slash and xml

https://devblogs.microsoft.com/cppblog/doxygen-and-xml-doc-comment-support/

**Pull request merged**